### PR TITLE
Sasha/inaugurator for lightbits

### PR DESCRIPTION
**Pull request by sashas-lb**
@abel-lb @maor-lb please review

### CODE REVIEW

File: inaugurator/sh.py [hunk 1]
```diff
@@ -26,3 +26,10 @@ def run(* args, ** kwargs):
         print "Command '%s' failed: %d\n%s" % (args, cmdPipe.returncode, output)
         raise ex
     return output
+
```
Comment (abel-lb) [inline inaugurator/sh.py]: what happen to the naming conventions for the commits ?

File: inaugurator/sh.py [hunk 2]
```diff
@@ -26,3 +26,10 @@ def run(* args, ** kwargs):
...
+def has_tool(rootPath, tool):
+    try:
+        run("/usr/sbin/busybox chroot %s sh -c 'which %s'" % (rootPath, tool))
```
Comment (abel-lb) [inline inaugurator/sh.py]: consider adding run_busybox method

File: inaugurator/sh.py [hunk 3]
```diff
@@ -26,3 +26,10 @@ def run(* args, ** kwargs):
...
+        return True
+    except:
+        return False
```
Comment (abel-lb) [inline inaugurator/sh.py]: if we fail for something that is not related to the existent of the tool, shouldn't we raise the original exception ?

File: inaugurator/grub.py [hunk 4]
```diff
@@ -47,6 +47,21 @@ def setSerialDevices(serialDevices, destination):
             userSettingsFile.write("GRUB_CMDLINE_LINUX=\"%s\"\n" % (consoleConfiguration,))
 
 
+def grub_prefix(destination):
+    if sh.has_tool(destination, 'grub2-install'):
+        return 'grub2'
+    if sh.has_tool(destination, 'grub-install'):
+        return 'grub'
+    return None
```
Comment (abel-lb) [inline inaugurator/grub.py]: why none instead of raising an exception if you later assert on this ?

File: inaugurator/ceremony.py [hunk 5]
```diff
@@ -190,7 +191,9 @@ def _createBootAndInstallGrub(self, destination):
             logging.info("Installing GRUB2...")
             grub.install(self._targetDevice, destination)
             logging.info("Reading newly generated GRUB2 configuration file for later use...")
-            grubConfigFilename = os.path.join(destination, "boot", "grub2", "grub.cfg")
+            grub_prefix = grub.grub_prefix(destination)
+            assert grub_prefix is not None
```
Comment (abel-lb) [inline inaugurator/ceremony.py]: install already does this check, so why you double check ?

Comment (abel-lb) [inline inaugurator/ceremony.py]: cosmetic: wrap strings in meaningful defines( for boot, grub.cfg, grub2-install, grub-install, grub2, grub...)

File: inaugurator/osmose.py [hunk 6]
```diff
@@ -20,7 +20,11 @@ def __init__(self, destination, objectStores, withLocalObjectStore, ignoreDirs,
                 logging.info("Sybmolic link removed.")
             localObjectStore = os.path.join(osmosisDir, "objectstore")
             absoluteIgnoreDirs.append(localObjectStore)
-            objectStores = localObjectStore + ("+" + objectStores if objectStores else "")
+        elif localObjectStore is not None:
```
Comment (abel-lb) [inline inaugurator/osmose.py]: consider logging all params in line #11

File: inaugurator/osmose.py [hunk 7]
```diff
@@ -20,7 +20,11 @@ def __init__(self, destination, objectStores, withLocalObjectStore, ignoreDirs,
...
+            localObjectStore = os.path.join(localObjectStore, "objectstore")
```
Comment (abel-lb) [inline inaugurator/osmose.py]: consider validating withLocalObjectStore exists and log some warning/error if not. Also  in this case avoid adding the path to objectStores

Comment (abel-lb) [inline inaugurator/osmose.py]: curious, if it's a separate partition, who will take care of mounting if not mounted ?

File: inaugurator/sh.py (hunk 1 above)
Comment (sashas-lb) [inline inaugurator/sh.py]: since this is forked project i do the same convention as in the rest of it

File: inaugurator/sh.py (hunk 2 above)
Comment (sashas-lb) [inline inaugurator/sh.py]: that does what ?

File: inaugurator/grub.py (hunk 4 above)
Comment (sashas-lb) [inline inaugurator/grub.py]: in case we will somehow support lilo oneday :) doubtgully

File: inaugurator/osmose.py (hunk 7 above)
Comment (sashas-lb) [inline inaugurator/osmose.py]: it is mounted in the beginning of the cycle

File: inaugurator/sh.py (hunk 1 above)
Comment (abel-lb) [inline inaugurator/sh.py]: IMHO we should use our naming conventions. BTW, you did use them but is some of the commits.

File: inaugurator/sh.py (hunk 2 above)
Comment (abel-lb) [inline inaugurator/sh.py]: receives the cmd, but without the need of specifying /usr/sbin/busybox chroot %s sh -c (the method does this by itself)

File: inaugurator/grub.py (hunk 4 above)
Comment (abel-lb) [inline inaugurator/grub.py]: ?!? then I hope grub_prefix will not be called :)

Comment (sashas-lb) [inline inaugurator/grub.py]: What?

Comment (abel-lb) [inline inaugurator/grub.py]: this is a grub specific method and lilo has nothing todo.. so don't understand why you wrote such comment. You should simple raise an exception.
